app/src/main/java/com/keylesspalace/tusky/viewdata/NotificationViewData.java

```diff
@@ -59,7 +59,7 @@ public Concrete(Notification.Type type, String id, Account account,
         }
 
         public Notification.Type getType() {
-            return type;
+            return type == null ? Notification.Type.UNKNOWN : type;
```

Comment:
New notification does not have a type at all? Why it's `null`?

Reply:
https://github.com/tuskyapp/Tusky/blob/08c1bbd25340269fdb173bb61ca4416a9d931eef/app/src/main/java/com/keylesspalace/tusky/entity/Notification.kt#L26

New notification type `poll` is not defined in Notification.Type so type will be null.

Reply:
Ah, I see. Maybe more correct would be to alter parser code so that it returns `UNKNOWN` if it doesn't match any.
Right now it makes a really confusing API in the data layer (we can have both `null` and `UNKNOWN`).
We could mark type field as `@JsonAdapter` and write our adapter (I can help with that). It could take 10 more minutes but it will be future-proof IMO.

Reply:
I think its even possible without a `JsonAdapter`, have a look how we handle `Visibility` in `Status.kt`

Reply:
When I passed instance original visibility `unleakable`, app crashed with same error.
So I think we need `JsonAdapter`...

```
2019-03-15 16:50:24.068 2655-2655/com.keylesspalace.tusky E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.keylesspalace.tusky, PID: 2655
    java.lang.NullPointerException: Attempt to invoke virtual method 'int com.keylesspalace.tusky.entity.Status$Visibility.ordinal()' on a null object reference
```

Reply:
Added `NotificationTypeAdapter` and anotated `Notification.Type` by `JsonAdapter`.
Please check whether it works correctly.